cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/GeneratedKubernetesResource.scala

```diff
@@ -16,10 +16,22 @@
 
 package com.lightbend.rp.reactivecli.runtime.kubernetes
 
-import argonaut.Json
+import argonaut._
 import com.lightbend.rp.reactivecli.runtime.GeneratedResource
+import Argonaut._
+import com.lightbend.rp.reactivecli.process.jq
 
 /**
  * Base type which represents generated Kubernetes resource.
  */
-private[reactivecli] trait GeneratedKubernetesResource extends GeneratedResource[Json]
+private[reactivecli] trait GeneratedKubernetesResource extends GeneratedResource[Json] {
+  def json: Json
+  def jqExpression: Option[String]
+
+  def payload: Json = jqExpression.fold(json)(
+    jq(_, json.nospaces)
+      .parse
+      .fold(
+        error => throw new RuntimeException(s"Unable to parse output from jq: $error"),
+        identity))
```

Comment:
This is probably a more style related concern, but I'd extract the `jq` call from this trait to the callsite where the resource is being generated.